net.bobbo.player-controller/player_scripts/crouch.gd

```diff
@@ -14,30 +14,60 @@ var crouch_current: float = 0
 var crouch_target: float = 0
 const crouch_action: String = "player_crouch"
 
-## default_player_height is needed for proper uncrouching height calculations 
+## _default_player_height is needed for proper uncrouching height calculations 
 ## if the maximum player height were to change during gameplay at any point this 
 ## value would need to somehow be updated to account for that default is 2 
 ## because the default height in player_controller is 2
 var _default_player_height: float = 2
-var _should_be_crouched: bool = false
+
+var _ray_cast: RayCast3D = null
 
 const _standing_height_multiplier: float = 1
 const _crouching_height_multiplier: float = 0.5
 
 #
 #	Functions
-#
+# 
 
 func player_ready():
 	_default_player_height = player.height
+	
+	_ray_cast = RayCast3D.new()
```

Comment:
This is interesting - creating a new raycast3d node at runtime. I think I'd prefer that the player behavior scene have this node already in it, because then when a non-coder inspects the scene itself, it makes it clear _how_ we are checking for a ceiling collision.. though this is not necessary to change. 

The benefit of creating a node like this is that you don't have to deal with setting up the scene. Not sure what is better yet.

Reply:
Turns out Godot did not like me doing this a bit so I had to fiddle around with referencing the node a bit but it should be good now